src/dune_engine/build_system.mli

```diff
@@ -108,6 +109,14 @@ module Error : sig
 
   val promotion : t -> Diff_promotion.Annot.t option
 
+  val make
+    :  id:Id.t
+    -> description:[ `Exn of Exn_with_backtrace.t | `Diagnostic of Compound_user_error.t ]
+    -> ?dir:Path.t
+    -> ?promotion:Diff_promotion.Annot.t
```

Comment:
Don't use optional arguments without a concrete reason.

Also, this function should be moved to a `For_tests` submodule.